From 5deeb26b207b63ea9c75531e8f084879a62e9b2f Mon Sep 17 00:00:00 2001
From: Joe Groocock <me@frebib.net>
Date: Sun, 12 Jan 2020 16:14:02 +0000
Subject: api: log respect x-forwarded-for

---
 logger/handler.go | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/logger/handler.go b/logger/handler.go
index b1b4a94b..609c5a29 100644
--- a/logger/handler.go
+++ b/logger/handler.go
@@ -35,10 +35,16 @@ func Middleware(next http.Handler) http.Handler {
 		start := time.Now()
 		next.ServeHTTP(w, r.WithContext(ctx))
 		end := time.Now()
+
+		client := r.RemoteAddr
+		fwded := r.Header.Get("X-Forwarded-For")
+		if fwded != "" {
+			client = fwded
+		}
 		log.WithFields(logrus.Fields{
 			"method":  r.Method,
 			"request": r.RequestURI,
-			"remote":  r.RemoteAddr,
+			"remote":  client,
 			"latency": end.Sub(start),
 			"time":    end.Format(time.RFC3339),
 		}).Debug()
-- 
2.36.1

